src/lib/passwordWithValidate.svelte

```diff
@@ -0,0 +1,178 @@
+    const indicatorStatus = {
+        none: "",
+        warning: "!border-warning-400",
+        error: "!border-error-600",
+    };
+
+    /**
+     * Password verification: Must...
+     * - contain min. 1: lowercase english letter (a-z)
+     * - contain min. 1: uppercase english letter (A-Z)
+     * - contain min. 1: number (0-9)
+     * - contain min. 1: special character
+     * - be min. 12 characters total
+     */
+    const passwordRegexPattern =
+        /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?\W).{12,}$/;
+
+    // Will get value "error" or "warning" according to validators
+    let passwordIndicator = indicatorStatus.none;
+    let passwordValidateIndicator = indicatorStatus.none;
+
+    // Formfield binding references
+    let passwordRefernce;
+    let passwordValidateReference;
+
+    let showPassword = false;
+    let showPasswordValidate = false;
+    const btnIcon = {
+        locked: "🔐",
+        unlocked: "🔓",
+    };
+
+    // External reference to password input
+    export let password;
+
+    // Functions
+    function checkPasswordEquality() {
+        if (passwordRefernce.value !== passwordValidateReference.value) {
+            passwordValidateIndicator = indicatorStatus.error;
+            return false;
+        }
+        // Clear previous errors
+        passwordValidateIndicator = indicatorStatus.none;
+        return true;
+    }
+
+    export function validatePassword() {
+        return passwordRegexPattern.test(passwordRefernce.value);
+    }
+
+    /**
+     * Removes password warnings or errors when any warnings existed
+     * and are no longer valid.
+     */
+    function rmPasswordWarningsOnValid() {
+        if (
+            passwordIndicator === indicatorStatus.warning &&
+            validatePassword()
+        ) {
+            passwordIndicator = indicatorStatus.none;
+        }
+        if (
+            passwordValidateIndicator === indicatorStatus.error &&
+            checkPasswordEquality()
+        ) {
+            passwordValidateIndicator = indicatorStatus.none;
+        }
+    }
+
+    /**
+     * Cheks if the password matches specific requirements of
+     * password regex. Additionally checks if both passwords
+     * are equal.
+     * Warnings will be dislayed when the user input is not correct.
+     * After calling this the password variable will be set to the
+     * entered value, indepedant wheather the input is valid or not.
+     * @returns boolean - true, if all inputs are valid
+     * @see passwordRegexPattern
+     * @see password
+     */
+    export const validateInput = () => {
+        // Save the password
+        password = passwordRefernce.value;
+
+        if (!validatePassword()) {
+            passwordIndicator = indicatorStatus.warning;
+            return false;
+        }
+
+        passwordIndicator = indicatorStatus.none;
+
+        if (!checkPasswordEquality()) {
+            return false; // Passwords are not equal
+        }
+
+        return true;
```

Comment:
```suggestion
        return checkPasswordEquality();
```
Sollte das nicht dasselbe machen?

Reply:
Hahah, ja das kann man sich da sparen 😆